owncloudApp/src/main/java/com/owncloud/android/presentation/providers/sharing/UsersAndGroupsSearchProvider.kt

```diff
@@ -180,6 +179,9 @@ class UsersAndGroupsSearchProvider : ContentProvider() {
             try {
                 while (namesIt.hasNext()) {
                     item = namesIt.next()
+                    if (item.label == AccountUtils.getUsernameOfAccount(account.name) && item.shareType == ShareType.USER) {
+                        item = namesIt.next()
```

Comment:
Please, don't do this.

If we use an iterator to iterate in a while loop, the condition that checks if we need to exit the loop is `hasNext()`. So, we use`next()` only when we are sure it `hasNext()`. If we want to proceed with the next step, [we use `continue`](https://kotlinlang.org/docs/returns.html)